twisted/scripts/_twistd_unix.py

```diff
@@ -198,6 +199,38 @@ def preApplication(self):
         self.oldstderr = sys.stderr
 
 
+    def _formatException(self, exception):
+        """
+        Format the C{exception} in preparation for writing to the
+        status pipe.  This does the right thing on Python 2 if the
+        exception's message is Unicode, and in all cases limits the
+        length of the message, prior to encoding, to 100 codepoints.
+
+        This means the return message may be longer than 100 bytes.
```

Comment:
I think the right thing to do is to change [the test below](https://github.com/twisted/twisted/pull/476#discussion-diff-73857170R1860) to verify that a truncated message (e.g., `b'Exception: blah \u20'`) works.  I also think the writer should truncate the message *after* encoding.  That way we *know* only 100 bytes will be written in the pipe.  I want to say it's impossible for 100 code points to end up being longer than `PIPE_BUF`, but I'd rather rewrite the code to be sure that's the case.  Otherwise there's a risk the writer will block, which would be terrible!